src/main/java/com/frameworkium/core/ui/pages/Visibility.java

```diff
@@ -0,0 +1,237 @@
+package com.frameworkium.core.ui.pages;
+
+import com.frameworkium.core.ui.annotations.*;
+import com.frameworkium.core.ui.driver.WebDriverWrapper;
+import com.frameworkium.core.ui.tests.BaseTest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.*;
+import ru.yandex.qatools.htmlelements.element.HtmlElement;
+import ru.yandex.qatools.htmlelements.element.TypifiedElement;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOf;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfAllElements;
+import static ru.yandex.qatools.htmlelements.utils.HtmlElementUtils.*;
+
+/**
+ * All things Frameworkium-related dealing with PageObject element visibility.
+ */
+public final class Visibility {
+
+    private static final List<Class<? extends Annotation>> VISIBILITY_ANNOTATION_CLASSES =
+            Arrays.asList(Visible.class, Invisible.class, ForceVisible.class);
+
+    private final Wait<WebDriver> wait;
+    private final JavascriptExecutor driver;
+
+    /**
+     * Uses the {@link Wait} and {@link WebDriverWrapper} from {@link BaseTest}.
+     */
+    public Visibility() {
+        this(BaseTest.getWait(), BaseTest.getDriver());
+    }
+
+    /**
+     * @param wait   a specific wait to use.
+     * @param driver a specific {@link JavascriptExecutor} to use.
+     */
+    public Visibility(Wait<WebDriver> wait, JavascriptExecutor driver) {
+        this.wait = wait;
+        this.driver = driver;
+    }
+
+    /**
+     * Custom wait which fills the gap left by Selenium whereby
+     * <code>not({@link ExpectedConditions#visibilityOf(WebElement)})</code>
+     * will fail if the element is not present, but
+     * {@link ExpectedConditions#invisibilityOfElementLocated(By)}
+     * waits for either not visibility or not present.
+     *
+     * @param element the element to wait for
+     * @return an {@link ExpectedCondition} which returns <strong>false</strong>
+     * iff the element is visible, otherwise <strong>true</strong>.
+     */
+    public static ExpectedCondition<Boolean> notPresentOrInvisible(
+            WebElement element) {
+        return driver -> {
+            try {
+                return !element.isDisplayed();
```

Comment:
will it? I'd have thought that the elementNotFoundException would be thrown almost instantly, since we've not asked it to wait for the element anywhere?